### PR TITLE
feat: add sftp check

### PR DESCRIPTION
**Pull request by Kaitou786**

### CODE REVIEW

File: checks/folder_smb.go [hunk 1]
```diff
@@ -141,6 +113,7 @@ func getServerDetails(serverPath string) (server, sharename, searchPath string,
 	case 2:
 		sharename = serverDetails[1]
 		searchPath = "."
+		fmt.Println("returning from here")
```
Comment (yashmehrotra) [inline checks/folder_smb.go:116]: I think you missed removing this

File: fixtures/datasources/SFTP/sftp_pass.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+apiVersion: canaries.flanksource.com/v1
+kind: Canary
+metadata:
+  name: sftp-pass
+spec:
+  interval: 30
+  folder:
+    - path: /Users/tarun/flanksource/canary-checker
```
Comment (yashmehrotra) [inline fixtures/datasources/SFTP/sftp_pass.yaml:8]: you can use `/tmp` for a generic path that exists everywhere

File: checks/folder_sftp.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+func sshConnect(host string, port int, user string, password string) (*ssh.Client, error) {
+	config := &ssh.ClientConfig{
+		User: user,
+		Auth: []ssh.AuthMethod{
+			ssh.Password(password),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+	client, err := ssh.Dial("tcp", host+":"+strconv.Itoa(port), config)
```
Comment (yashmehrotra) [inline checks/folder_sftp.go:55]: (nitpick: can-ignore):
`client, err := ssh.Dial("tcp", fmt.Sprintf("%s:%d", host, port), config)`

File: checks/folder_sftp.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	if err != nil {
+		return nil, err
+	}
+	return client, nil
```
Comment (yashmehrotra) [inline checks/folder_sftp.go:59]: Can just `return ssh.Dial(...)`

File: checks/folder_smb.go (hunk 1 above)
Comment (Kaitou786) [reply checks/folder_smb.go]: yeah xD

File: fixtures/datasources/SFTP/sftp_pass.yaml (hunk 2 above)
Comment (Kaitou786) [reply fixtures/datasources/SFTP/sftp_pass.yaml]: Yeah updated... just added for a quick check

File: checks/folder_sftp.go (hunk 3 above)
Comment (Kaitou786) [reply checks/folder_sftp.go]: Updated

File: checks/folder_sftp.go (hunk 4 above)
Comment (Kaitou786) [reply checks/folder_sftp.go]: updated

File: checks/folder_check.go [hunk 5]
```diff
@@ -9,17 +9,21 @@ import (
...
-	Oldest  os.FileInfo
-	Newest  os.FileInfo
-	MinSize os.FileInfo
-	MaxSize os.FileInfo
-	Files   []os.FileInfo
+	Oldest                os.FileInfo
+	Newest                os.FileInfo
+	MinSize               os.FileInfo
+	MaxSize               os.FileInfo
+	SupportsTotalSize     bool
```
Comment (moshloop) [inline checks/folder_check.go:22]: This would part of the interface definition, each implementation would return a value on whether they support it or not

Comment (Kaitou786) [reply checks/folder_check.go:22]: We don't have a standard interface currently used by all the folder checks?

We can make a function on the Connection though?

**Pull request merged**